src/general/disallowanonymous/src.rego

```diff
@@ -1,19 +1,34 @@
 package k8sdisallowanonymous
 
+
+violation[{"msg": msg}] {
+  not is_allowed(input.review.object.roleRef, object.get(input, ["parameters", "allowedRoles"], []))
+
+  group := ["system:unauthenticated", "system:anonymous"][_]
+  subject_is(input.review.object.subjects[_], group)
+
+  msg := message(group)
+}
+
 violation[{"msg": msg}] {
   not is_allowed(input.review.object.roleRef, object.get(input, ["parameters", "allowedRoles"], []))
-  review(input.review.object.subjects[_])
-  msg := sprintf("Unauthenticated user reference is not allowed in %v %v ", [input.review.object.kind, input.review.object.metadata.name])
+
+  object.get(input, ["parameters", "disallowAuthenticated"], false)
+
+  group := "system:authenticated"
+  subject_is(input.review.object.subjects[_], group)
+
+  msg := message(group)
 }
 
 is_allowed(role, allowedRoles) {
   role.name == allowedRoles[_]
 }
 
-review(subject) = true {
-  subject.name == "system:unauthenticated"
+subject_is(subject, expected) {
+  subject.name == expected
 }
 
-review(subject) = true {
-  subject.name == "system:anonymous"
+message(group) := val {
+  val := sprintf("%v is not allowed as a subject name in %v %v", [group, input.review.object.kind, input.review.object.metadata.name])
```

Comment:
Just calling out that by including `group` we are slightly changing the original behavior.  As it's just the message I don't think it a breaking change, but wanted to note that is some cases it will increases the number of violations as they are now group specific.

Reply:
Good eye!

I confirmed with Max that going from 1 to multiple violations is not a breaking change.

My thinking was that before:

1. Someone with `system:anonymous` as the subject would get a message saying that `Unuthenticated` was the problem.

2. Someone with a binding to _both_ `system:anonymous` AND `system:unauthenticated` would get only a single violation.  Once they fix that violation, they'd then get _another_ violation for the other binding.  This UX is confusing.

Reply:
Yup, agreed, and makes sense to me!